[BUG] Uploading PNG file results in file not found

### Description

I have a screenshot file (PNG) that has a size of 38 MB and a dimension of 2560x27288, so a fairly big PNG.

After successful upload, Paperless starts consuming the file and suddenly quits with the following error: `chat.png: Error occurred while consuming document chat.png: InputFileError: File not found - /tmp/paperless/paperless-vfmktwuq/image-no-alpha`

### Steps to reproduce

1. Uploading file by dropping it in the GUI.
2. Wait until file is uploaded and Paperless to start consume.
3. Error pops up.

### Webserver logs

```bash
[2024-04-03 08:34:26,931] [DEBUG] [paperless.tasks] Skipping plugin CollatePlugin

[2024-04-03 08:34:26,931] [DEBUG] [paperless.tasks] Skipping plugin BarcodePlugin

[2024-04-03 08:34:26,931] [DEBUG] [paperless.tasks] Executing plugin WorkflowTriggerPlugin

[2024-04-03 08:34:26,959] [INFO] [paperless.tasks] WorkflowTriggerPlugin completed with no message

[2024-04-03 08:34:27,056] [INFO] [paperless.consumer] Consuming chat.png

[2024-04-03 08:34:27,082] [DEBUG] [paperless.consumer] Detected mime type: image/png

[2024-04-03 08:34:27,086] [INFO] [paperless.consumer] Executing pre-consume script /opt/paperless/scripts/removepassword.py

[2024-04-03 08:34:27,356] [INFO] [paperless.consumer] /opt/paperless/scripts/removepassword.py exited 0

[2024-04-03 08:34:27,405] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser

[2024-04-03 08:34:27,408] [DEBUG] [paperless.consumer] Parsing chat.png...

[2024-04-03 08:34:28,825] [WARNING] [paperless.parsing.tesseract] Error while getting DPI from image /tmp/paperless/paperless-ngxbpyj79li/chat.png: 'dpi'

[2024-04-03 08:34:28,825] [DEBUG] [paperless.parsing.tesseract] Estimated DPI 309 based on image width 2560

[2024-04-03 08:34:28,826] [INFO] [paperless.parsing.tesseract] Removing alpha layer from /tmp/paperless/paperless-ngxbpyj79li/chat.png for compatibility with img2pdf

[2024-04-03 08:34:29,648] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': PosixPath('/tmp/paperless/paperless-vfmktwuq/image-no-alpha'), 'output_file': PosixPath('/tmp/paperless/paperless-vfmktwuq/archive.pdf'), 'use_threads': True, 'jobs': 2, 'language': 'eng', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': 'RGB', 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-vfmktwuq/sidecar.txt'), 'image_dpi': 309}

[2024-04-03 08:34:32,947] [WARNING] [paperless.parsing.tesseract] Encountered an error while running OCR: File not found - /tmp/paperless/paperless-vfmktwuq/image-no-alpha. Attempting force OCR to get the text.

[2024-04-03 08:34:32,947] [WARNING] [paperless.parsing.tesseract] Error while getting DPI from image /tmp/paperless/paperless-ngxbpyj79li/chat.png: 'dpi'

[2024-04-03 08:34:32,947] [DEBUG] [paperless.parsing.tesseract] Estimated DPI 309 based on image width 2560

[2024-04-03 08:34:32,948] [INFO] [paperless.parsing.tesseract] Removing alpha layer from /tmp/paperless/paperless-ngxbpyj79li/chat.png for compatibility with img2pdf

[2024-04-03 08:34:32,954] [DEBUG] [paperless.parsing.tesseract] Fallback: Calling OCRmyPDF with args: {'input_file': PosixPath('/tmp/paperless/paperless-vfmktwuq/image-no-alpha'), 'output_file': PosixPath('/tmp/paperless/paperless-vfmktwuq/archive-fallback.pdf'), 'use_threads': True, 'jobs': 2, 'language': 'eng', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': 'RGB', 'force_ocr': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-vfmktwuq/sidecar-fallback.txt'), 'image_dpi': 309}

[2024-04-03 08:34:33,057] [ERROR] [paperless.consumer] Error occurred while consuming document chat.png: InputFileError: File not found - /tmp/paperless/paperless-vfmktwuq/image-no-alpha

Traceback (most recent call last):

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/_validation.py", line 282, in create_input_file

    safe_symlink(options.input_file, target)

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/helpers.py", line 171, in safe_symlink

    raise FileNotFoundError(f"trying to create a broken symlink to {input_file}")

FileNotFoundError: trying to create a broken symlink to /tmp/paperless/paperless-vfmktwuq/image-no-alpha

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/opt/paperless/src/paperless_tesseract/parsers.py", line 363, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/api.py", line 375, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/_pipelines/ocr.py", line 225, in run_pipeline

    return _run_pipeline(options, plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/_pipelines/ocr.py", line 168, in _run_pipeline

    start_input_file, original_filename = create_input_file(options, work_folder)

                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/_validation.py", line 305, in create_input_file

    raise InputFileError(msg) from e

ocrmypdf.exceptions.InputFileError: File not found - /tmp/paperless/paperless-vfmktwuq/image-no-alpha

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/_validation.py", line 282, in create_input_file

    safe_symlink(options.input_file, target)

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/helpers.py", line 171, in safe_symlink

    raise FileNotFoundError(f"trying to create a broken symlink to {input_file}")

FileNotFoundError: trying to create a broken symlink to /tmp/paperless/paperless-vfmktwuq/image-no-alpha

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/opt/paperless/src/paperless_tesseract/parsers.py", line 414, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/api.py", line 375, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/_pipelines/ocr.py", line 225, in run_pipeline

    return _run_pipeline(options, plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/_pipelines/ocr.py", line 168, in _run_pipeline

    start_input_file, original_filename = create_input_file(options, work_folder)

                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/dist-packages/ocrmypdf/_validation.py", line 305, in create_input_file

    raise InputFileError(msg) from e

ocrmypdf.exceptions.InputFileError: File not found - /tmp/paperless/paperless-vfmktwuq/image-no-alpha

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/usr/local/lib/python3.11/dist-packages/asgiref/sync.py", line 349, in main_wrap

    raise exc_info[1]

  File "/opt/paperless/src/documents/consumer.py", line 516, in try_consume_file

    document_parser.parse(self.working_copy, mime_type, self.filename)

  File "/opt/paperless/src/paperless_tesseract/parsers.py", line 426, in parse

    raise ParseError(f"{e.__class__.__name__}: {e!s}") from e

documents.parsers.ParseError: InputFileError: File not found - /tmp/paperless/paperless-vfmktwuq/image-no-alpha

Drop files to begin upload
```


### Browser logs

_No response_

### Paperless-ngx version

2.6.3

### Host OS

Debian 12 x86_64

### Installation method

Bare metal

### Browser

_No response_

### Configuration changes

_No response_

### Other

Installed via Ttecks Proxmox helper scripts.

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/documents/consumer.py
+++ b/src/documents/consumer.py
@@ -5,8 +5,6 @@ import tempfile
 import uuid
 from enum import Enum
 from pathlib import Path
-from subprocess import CompletedProcess
-from subprocess import run
 from typing import TYPE_CHECKING
 from typing import Optional
 
@@ -51,6 +49,7 @@ from documents.signals import document_consumption_finished
 from documents.signals import document_consumption_started
 from documents.utils import copy_basic_file_stats
 from documents.utils import copy_file_with_basic_stats
+from documents.utils import run_subprocess
 
 
 class WorkflowTriggerPlugin(
@@ -397,19 +396,14 @@ class Consumer(LoggingMixin):
         script_env["TASK_ID"] = self.task_id or ""
 
         try:
-            completed_proc = run(
-                args=[
+            run_subprocess(
+                [
                     settings.PRE_CONSUME_SCRIPT,
                     original_file_path,
                 ],
-                env=script_env,
-                capture_output=True,
+                script_env,
+                self.log,
             )
-
-            self._log_script_outputs(completed_proc)
-
-            # Raises exception on non-zero output
-            completed_proc.check_returncode()
 
         except Exception as e:
             self._fail(
@@ -468,8 +462,8 @@ class Consumer(LoggingMixin):
         script_env["TASK_ID"] = self.task_id or ""
 
         try:
-            completed_proc = run(
-                args=[
+            run_subprocess(
+                [
                     settings.POST_CONSUME_SCRIPT,
                     str(document.pk),
                     document.get_public_filename(),
@@ -480,14 +474,9 @@ class Consumer(LoggingMixin):
                     str(document.correspondent),
                     str(",".join(document.tags.all().values_list("name", flat=True))),
                 ],
-                env=script_env,
-                capture_output=True,
+                script_env,
+                self.log,
             )
-
-            self._log_script_outputs(completed_proc)
-
-            # Raises exception on non-zero output
-            completed_proc.check_returncode()
 
         except Exception as e:
             self._fail(
@@ -928,41 +917,6 @@ class Consumer(LoggingMixin):
             copy_basic_file_stats(source, target)
         except Exception:  # pragma: no cover
             pass
-
-    def _log_script_outputs(self, completed_process: CompletedProcess):
-        """
-        Decodes a process stdout and stderr streams and logs them to the main log
-        """
-        # Log what the script exited as
-        self.log.info(
-            f"{completed_process.args[0]} exited {completed_process.returncode}",
-        )
-
-        # Decode the output (if any)
-        if len(completed_process.stdout):
-            stdout_str = (
-                completed_process.stdout.decode("utf8", errors="ignore")
-                .strip()
-                .split(
-                    "\n",
-                )
-            )
-            self.log.info("Script stdout:")
-            for line in stdout_str:
-                self.log.info(line)
-
-        if len(completed_process.stderr):
-            stderr_str = (
-                completed_process.stderr.decode("utf8", errors="ignore")
-                .strip()
-                .split(
-                    "\n",
-                )
-            )
-
-            self.log.warning("Script stderr:")
-            for line in stderr_str:
-                self.log.warning(line)
 
 
 def parse_doc_title_w_placeholders(

--- a/src/documents/converters.py
+++ b/src/documents/converters.py
@@ -1,5 +1,4 @@
 from pathlib import Path
-from subprocess import run
 
 import img2pdf
 from django.conf import settings
@@ -7,6 +6,7 @@ from PIL import Image
 
 from documents.utils import copy_basic_file_stats
 from documents.utils import maybe_override_pixel_limit
+from documents.utils import run_subprocess
 
 
 def convert_from_tiff_to_pdf(tiff_path: Path, target_directory: Path) -> Path:
@@ -27,7 +27,7 @@ def convert_from_tiff_to_pdf(tiff_path: Path, target_directory: Path) -> Path:
         # Note the save into the temp folder, so as not to trigger a new
         # consume
         scratch_image = target_directory / tiff_path.name
-        run(
+        run_subprocess(
             [
                 settings.CONVERT_BINARY,
                 "-alpha",

--- a/src/documents/parsers.py
+++ b/src/documents/parsers.py
@@ -169,7 +169,7 @@ def run_convert(
         run_subprocess(args, environment, logger)
     except subprocess.CalledProcessError as e:
         raise ParseError(f"Convert failed at {args}") from e
-    except Exception as e:
+    except Exception as e:  # pragma: no cover
         raise ParseError("Unknown error running convert") from e
 
 

--- a/src/documents/parsers.py
+++ b/src/documents/parsers.py
@@ -18,6 +18,7 @@ from django.utils import timezone
 from documents.loggers import LoggingMixin
 from documents.signals import document_consumer_declaration
 from documents.utils import copy_file_with_basic_stats
+from documents.utils import run_subprocess
 
 # This regular expression will try to find dates in the document at
 # hand and will match the following formats:
@@ -164,8 +165,12 @@ def run_convert(
 
     logger.debug("Execute: " + " ".join(args), extra={"group": logging_group})
 
-    if not subprocess.Popen(args, env=environment).wait() == 0:
-        raise ParseError(f"Convert failed at {args}")
+    try:
+        run_subprocess(args, environment, logger)
+    except subprocess.CalledProcessError as e:
+        raise ParseError(f"Convert failed at {args}") from e
+    except Exception as e:
+        raise ParseError("Unknown error running convert") from e
 
 
 def get_default_thumbnail() -> Path:
@@ -188,9 +193,12 @@ def make_thumbnail_from_pdf_gs_fallback(in_path, temp_dir, logging_group=None) -
     # Ghostscript doesn't handle WebP outputs
     gs_out_path = os.path.join(temp_dir, "gs_out.png")
     cmd = [settings.GS_BINARY, "-q", "-sDEVICE=pngalpha", "-o", gs_out_path, in_path]
+
     try:
-        if not subprocess.Popen(cmd).wait() == 0:
-            raise ParseError(f"Thumbnail (gs) failed at {cmd}")
+        try:
+            run_subprocess(cmd, logger=logger)
+        except subprocess.CalledProcessError as e:
+            raise ParseError(f"Thumbnail (gs) failed at {cmd}") from e
         # then run convert on the output from gs to make WebP
         run_convert(
             density=300,

--- a/src/documents/tests/test_consumer.py
+++ b/src/documents/tests/test_consumer.py
@@ -934,7 +934,7 @@ class PreConsumeTestCase(TestCase):
 
         return super().setUp()
 
-    @mock.patch("documents.consumer.run")
+    @mock.patch("documents.consumer.run_subprocess")
     @override_settings(PRE_CONSUME_SCRIPT=None)
     def test_no_pre_consume_script(self, m):
         c = Consumer()
@@ -942,7 +942,7 @@ class PreConsumeTestCase(TestCase):
         c.run_pre_consume_script()
         m.assert_not_called()
 
-    @mock.patch("documents.consumer.run")
+    @mock.patch("documents.consumer.run_subprocess")
     @mock.patch("documents.consumer.Consumer._send_progress")
     @override_settings(PRE_CONSUME_SCRIPT="does-not-exist")
     def test_pre_consume_script_not_found(self, m, m2):
@@ -951,7 +951,7 @@ class PreConsumeTestCase(TestCase):
         c.working_copy = "path-to-file"
         self.assertRaises(ConsumerError, c.run_pre_consume_script)
 
-    @mock.patch("documents.consumer.run")
+    @mock.patch("documents.consumer.run_subprocess")
     def test_pre_consume_script(self, m):
         with tempfile.NamedTemporaryFile() as script:
             with override_settings(PRE_CONSUME_SCRIPT=script.name):
@@ -963,10 +963,10 @@ class PreConsumeTestCase(TestCase):
 
                 m.assert_called_once()
 
-                args, kwargs = m.call_args
+                args, _ = m.call_args
 
-                command = kwargs["args"]
-                environment = kwargs["env"]
+                command = args[0]
+                environment = args[1]
 
                 self.assertEqual(command[0], script.name)
                 self.assertEqual(command[1], "path-to-file")
@@ -1050,7 +1050,7 @@ class PostConsumeTestCase(TestCase):
 
         return super().setUp()
 
-    @mock.patch("documents.consumer.run")
+    @mock.patch("documents.consumer.run_subprocess")
     @override_settings(POST_CONSUME_SCRIPT=None)
     def test_no_post_consume_script(self, m):
         doc = Document.objects.create(title="Test", mime_type="application/pdf")
@@ -1075,7 +1075,7 @@ class PostConsumeTestCase(TestCase):
             doc,
         )
 
-    @mock.patch("documents.consumer.run")
+    @mock.patch("documents.consumer.run_subprocess")
     def test_post_consume_script_simple(self, m):
         with tempfile.NamedTemporaryFile() as script:
             with override_settings(POST_CONSUME_SCRIPT=script.name):
@@ -1085,7 +1085,7 @@ class PostConsumeTestCase(TestCase):
 
                 m.assert_called_once()
 
-    @mock.patch("documents.consumer.run")
+    @mock.patch("documents.consumer.run_subprocess")
     def test_post_consume_script_with_correspondent(self, m):
         with tempfile.NamedTemporaryFile() as script:
             with override_settings(POST_CONSUME_SCRIPT=script.name):
@@ -1106,10 +1106,10 @@ class PostConsumeTestCase(TestCase):
 
                 m.assert_called_once()
 
-                _, kwargs = m.call_args
+                args, _ = m.call_args
 
-                command = kwargs["args"]
-                environment = kwargs["env"]
+                command = args[0]
+                environment = args[1]
 
                 self.assertEqual(command[0], script.name)
                 self.assertEqual(command[1], str(doc.pk))

--- a/src/documents/utils.py
+++ b/src/documents/utils.py
@@ -1,6 +1,9 @@
+import logging
 import shutil
 from os import utime
 from pathlib import Path
+from subprocess import CompletedProcess
+from subprocess import run
 from typing import Optional
 from typing import Union
 
@@ -56,3 +59,54 @@ def maybe_override_pixel_limit() -> None:
         if pixel_count == 0:
             pixel_count = None
         Image.MAX_IMAGE_PIXELS = pixel_count
+
+
+def run_subprocess(
+    arguments: list[str],
+    env: dict[str, str] | None = None,
+    logger: logging.Logger | None = None,
+    *,
+    check_exit_code: bool = True,
+    log_stdout: bool = True,
+    log_stderr: bool = True,
+) -> CompletedProcess:
+    """
+    Runs a subprocess and logs its output, checking return code if requested
+    """
+
+    proc_name = arguments[0]
+
+    completed_proc = run(args=arguments, env=env, capture_output=True, check=False)
+
+    if logger:
+        logger.info(f"{proc_name} exited {completed_proc.returncode}")
+
+    if log_stdout and logger and completed_proc.stdout:
+        stdout_str = (
+            completed_proc.stdout.decode("utf8", errors="ignore")
+            .strip()
+            .split(
+                "\n",
+            )
+        )
+        logger.info(f"{proc_name} stdout:")
+        for line in stdout_str:
+            logger.info(line)
+
+    if log_stderr and logger and completed_proc.stderr:
+        stderr_str = (
+            completed_proc.stderr.decode("utf8", errors="ignore")
+            .strip()
+            .split(
+                "\n",
+            )
+        )
+        logger.info(f"{proc_name} stderr:")
+        for line in stderr_str:
+            logger.warning(line)
+
+    # Last, if requested, after logging outputs
+    if check_exit_code:
+        completed_proc.check_returncode()
+
+    return completed_proc

--- a/src/documents/utils.py
+++ b/src/documents/utils.py
@@ -63,8 +63,8 @@ def maybe_override_pixel_limit() -> None:
 
 def run_subprocess(
     arguments: list[str],
-    env: dict[str, str] | None = None,
-    logger: logging.Logger | None = None,
+    env: Optional[dict[str, str]] = None,
+    logger: Optional[logging.Logger] = None,
     *,
     check_exit_code: bool = True,
     log_stdout: bool = True,

--- a/src/paperless_tesseract/parsers.py
+++ b/src/paperless_tesseract/parsers.py
@@ -1,6 +1,5 @@
 import os
 import re
-import subprocess
 import tempfile
 from pathlib import Path
 from typing import TYPE_CHECKING
@@ -13,6 +12,7 @@ from documents.parsers import DocumentParser
 from documents.parsers import ParseError
 from documents.parsers import make_thumbnail_from_pdf
 from documents.utils import maybe_override_pixel_limit
+from documents.utils import run_subprocess
 from paperless.config import OcrConfig
 from paperless.models import ArchiveFileChoices
 from paperless.models import CleanChoices
@@ -103,7 +103,7 @@ class RasterisedDocumentParser(DocumentParser):
 
     def remove_alpha(self, image_path: str) -> Path:
         no_alpha_image = Path(self.tempdir) / "image-no-alpha"
-        subprocess.run(
+        run_subprocess(
             [
                 settings.CONVERT_BINARY,
                 "-alpha",
@@ -111,6 +111,7 @@ class RasterisedDocumentParser(DocumentParser):
                 image_path,
                 no_alpha_image,
             ],
+            logger=self.log,
         )
         return no_alpha_image
 
@@ -169,7 +170,7 @@ class RasterisedDocumentParser(DocumentParser):
                 mode="w+",
                 dir=self.tempdir,
             ) as tmp:
-                subprocess.run(
+                run_subprocess(
                     [
                         "pdftotext",
                         "-q",
@@ -179,6 +180,7 @@ class RasterisedDocumentParser(DocumentParser):
                         pdf_file,
                         tmp.name,
                     ],
+                    logger=self.log,
                 )
                 text = self.read_file_handle_unicode_errors(Path(tmp.name))
 

